source/actors/player/PlayerHealth.hx

```diff
@@ -37,14 +37,17 @@ class PlayerHealth extends FlxTypedGroup<HeartContainer> {
     }
 
     public function updateGraphics(currentHealth:Float):Void {
-        for (val in 0...countLiving()) {
+        for (val in 0...length) {
+            trace("Health at call " + (val + 1) + ": " + currentHealth);
             var heart:HeartContainer = members[val];
             if (currentHealth >= HeartContainer.MAX_FRAG_AMOUNT) {
                 currentHealth -= HeartContainer.MAX_FRAG_AMOUNT;
                 heart.setFragments(HeartContainer.MAX_FRAG_AMOUNT);
-            } else {
+            } else if (currentHealth > 0) {
                 heart.setFragments(Std.int(currentHealth));
-                break;
+                currentHealth -= currentHealth;
```

Comment:
After re-reading over the code a couple of times, this makes sense (the only way we reach this conditional is if we're filling in the last partially filled heart) but it read very strangely to me during my first pass.

There's no need to change anything, but I do wish there was a way to make this a little easier to read. 

Maybe it'd be more straightforward (albeit involve a bit more calculation) if currentHealth didn't change during loop iterations? I think most of my confusion came from that. Maybe each cycle through the loop could just do something like

```
for (i in 0...length) {
    var healthToDisplay = currentHealth - i * HeartContainer.MAX_FRAG_AMOUNT 
    members[i].setFragments(Math.max(healthToDisplay, 0));
}
```

Reply:
That makes sense. I forgot I had a safeguard in the HeartContainer to take care of excess fragment value. I tested that out and it worked well and I agree it is more readable.